source/connect/csot.txt

```diff
@@ -1,210 +1,218 @@
+The following code examples use the ``timeoutMS`` option to specify a timeout of 10 seconds:
 
-pymongo.timeout Overrides timeoutMS
------------------------------------
+.. tabs::
 
-The ``~pymongo.timeout()`` method overrides ``timeoutMS``. Within a
-``~pymongo.timeout`` block, a client's ``timeoutMS`` option is ignored:
+   .. tab:: MongoClient
+      :tabid: mongoclient
 
-.. code-block:: python
+      .. code-block:: python
+         :emphasize-lines: 2
 
-   client = MongoClient("mongodb://localhost/?timeoutMS=10000")
-   coll = client.test.test
-   coll.insert_one({"name": "Nunu"})  # Uses the client's 10-second timeout.
-   # pymongo.timeout overrides the client's timeoutMS.
-   with pymongo.timeout(20):
-       coll.insert_one({"name": "Nunu"})  # Uses the 20-second timeout.
-   with pymongo.timeout(5):
-       coll.find_one({"name": "Nunu"})  # Uses the 5-second timeout.
+         client = pymongo.MongoClient("mongodb://<username>:<password>@<hostname@:<port>",
+                                      timeoutMS=10000)
 
-pymongo.timeout Is Thread-safe
-------------------------------
+   .. tab:: Connection String
+      :tabid: connectionstring
 
-The ``~pymongo.timeout()`` method is thread-safe. The timeout applies only to the current
-thread, and multiple threads can configure different timeouts in parallel.
+      .. code-block:: python
+         :emphasize-lines: 1
 
-pymongo.timeout Is asyncio-safe
--------------------------------
+         uri = "mongodb://<username>:<password>@<hostname@:<port>/?timeoutMS=10000"
+         client = pymongo.MongoClient(uri)
 
-The ``~pymongo.timeout()`` method is ``asyncio``-safe. The timeout applies only to the
-current Task, and multiple Tasks can configure different timeouts concurrently.
+If you specify the ``timeoutMS`` option, {+driver-short+} automatically applies the
+specified timeout to each server operation. The following code examples specify
+a timeout of 10 seconds, then call the ``insert_one()`` and
+``find_one()`` methods. To see how the ``timeoutMS`` option compares to using the
+``timeout()`` method, select the corresponding tab.
 
-The ``~pymongo.timeout()`` method can be used identically in
-`Motor <https://github.com/mongodb/motor>`__: 
+.. tabs::
 
-.. code-block:: python
+   .. tab:: timeoutMS Option
+      :tabid: uri-option
 
-   import motor.motor_asyncio
-   client = motor.motor_asyncio.AsyncIOMotorClient()
-   coll = client.test.test
-   with pymongo.timeout(10):
-       await coll.insert_one({"name": "Nunu"})
-       await coll.find_one({"name": "Nunu"})
+      .. code-block:: python
 
-Troubleshooting
----------------
+         uri = "mongodb://<username>:<password>@<hostname@:<port>/?timeoutMS=10000"
+         client = pymongo.MongoClient(uri)
 
-In code, you can identify timeout errors by using the ``pymongo.errors.PyMongoError.timeout``
-property. Some specific timeout errors examples are described below.
+         coll = client.test.test
```

Comment:
Accessing the collection like this didn't work for me when I tried it in another code example. I think this syntax is for shell. Should probably update it to be application code: i.e. `coll = test_database[test]` 

Applies to all code blocks